pkg/agent/dispatcher.go

```diff
@@ -373,11 +370,15 @@ func (disp *Dispatcher) run() {
 			handlers,
 		)
 		if err != nil {
-			logger.Error("error handling message -> panicking", zap.Error(err))
-			// TODO: fix this comment for the agent
-			// We actually want to panic here so we get respawned by the inittab,
-			// and so the monitor's connection is closed and it also gets restarted
-			panic(err)
+			func() {
+				logger.Error("error handling message -> triggering informant server exit", zap.Error(err))
+				disp.server.runner.lock.Lock()
+				defer disp.server.runner.lock.Unlock()
+				disp.server.exit(InformantServerExitStatus{
+					Err:            err,
+					RetryShouldFix: false,
+				})
+			}()
```

Comment:
I think that was just an oversight on my part; we should probably explicitly return.

For the second part, do you mean check `ctx.Cause` first and then exit?

Reply:
> do you mean check `ctx.Cause` first and then exit?

`ctx.Err()`, but yeah — basically, it might be worthwhile to do something like: "if err != nil, but the context is already canceled (i.e. ctx.Err() != nil), then the error _probably_ was expected, so just log it as a warning, and skip the `server.exit`"

Reply:
Ahh gotchu. For some reason I don't see the docs for that function (although it's coming up in nvim :thinking:)